### PR TITLE
[Merged by Bors] - feat: add lemmas to simplify List.modifyNth

### PR DESCRIPTION
**Pull request by BoltonBailey**
Adds three lemmas, I don't think any of them are in the library anywhere already.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/List/Basic.lean [hunk 1]
```diff
@@ -2209,6 +2209,18 @@ theorem modifyNthTail_eq_take_drop (f : List α → List α) (H : f [] = []) :
...
+theorem modifyNth_nil (f : α → α) (n) :
+    modifyNth f n [] = [] := by cases n <;> rfl
+
+@[simp]
+theorem modifyNth_zero_cons (f : α → α) (a l) :
+    modifyNth f 0 (a :: l) = f a :: l := rfl
+
+@[simp]
+theorem modifyNth_succ_cons (f : α → α) (n a l) :
+    modifyNth f (n.succ) (a :: l) = a :: modifyNth f n l := rfl
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/List/Basic.lean:2222]:
```suggestion
    modifyNth f n.succ (a :: l) = a :: modifyNth f n l := rfl
```
or should we use `n + 1`?

File: Mathlib/Data/List/Basic.lean [hunk 2]
```diff
@@ -2209,6 +2209,18 @@ theorem modifyNthTail_eq_take_drop (f : List α → List α) (H : f [] = []) :
   | n + 1, b :: l => congr_arg (cons b) (modifyNthTail_eq_take_drop f H n l)
 #align list.modify_nth_tail_eq_take_drop List.modifyNthTail_eq_take_drop
 
+@[simp]
+theorem modifyNth_nil (f : α → α) (n) :
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/List/Basic.lean:2213]: I would consider giving all arguments explicit types, even if lean can infer them

File: Mathlib/Data/List/Basic.lean (hunk 1 above)
Comment (BoltonBailey) [reply Mathlib/Data/List/Basic.lean]: Yes, ok the surrounding text uses n+1, so I'll change that

File: Mathlib/Data/List/Basic.lean (hunk 2 above)
Comment (BoltonBailey) [reply Mathlib/Data/List/Basic.lean:2213]: Ok sure

**Comment by Ruben-VandeVelde** [comment]
maintainer merge

**Pull request closed without merging**